From 7a93a3d8bfd7316053ad7535d6f32df3566c6e63 Mon Sep 17 00:00:00 2001
From: Wenlong Zhuang <daisen.zhuang@rock-chips.com>
Date: Wed, 7 Nov 2018 17:41:44 +0800
Subject: [PATCH] media: rockchip/cif: compatible with old dts config

Maybe not configure 'reg-names' property in dts file for px30/rk3128/rk3288.

Change-Id: I1f415baa51c13438237390794e8265e00bc29ee1
Signed-off-by: Wenlong Zhuang <daisen.zhuang@rock-chips.com>
---
 drivers/media/platform/rockchip/cif/dev.c | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/rockchip/cif/dev.c b/drivers/media/platform/rockchip/cif/dev.c
index ddefd4dd7356..beb7a5f54c5d 100644
--- a/drivers/media/platform/rockchip/cif/dev.c
+++ b/drivers/media/platform/rockchip/cif/dev.c
@@ -451,11 +451,6 @@ static int rkcif_plat_probe(struct platform_device *pdev)
 	dev_set_drvdata(dev, cif_dev);
 	cif_dev->dev = dev;
 
-	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "cif_regs");
-	cif_dev->base_addr = devm_ioremap_resource(dev, res);
-	if (IS_ERR(cif_dev->base_addr))
-		return PTR_ERR(cif_dev->base_addr);
-
 	irq = platform_get_irq(pdev, 0);
 	if (irq < 0)
 		return irq;
@@ -472,12 +467,23 @@ static int rkcif_plat_probe(struct platform_device *pdev)
 	cif_dev->chip_id = data->chip_id;
 	if (data->chip_id == CHIP_RK1808_CIF) {
 		using_pingpong = 1;
+
+		res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "cif_regs");
+		cif_dev->base_addr = devm_ioremap_resource(dev, res);
+		if (IS_ERR(cif_dev->base_addr))
+			return PTR_ERR(cif_dev->base_addr);
+
 		res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "csihost_regs");
 		cif_dev->csi_base = devm_ioremap_resource(dev, res);
 		if (IS_ERR(cif_dev->csi_base))
 			return PTR_ERR(cif_dev->csi_base);
 	} else {
 		using_pingpong = 0;
+
+		res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+		cif_dev->base_addr = devm_ioremap_resource(dev, res);
+		if (IS_ERR(cif_dev->base_addr))
+			return PTR_ERR(cif_dev->base_addr);
 	}
 
 	if (data->clks_num > RKCIF_MAX_BUS_CLK ||
-- 
2.35.3

